Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/RTCVideoDecoderH264.mm

```diff
@@ -76,7 +105,7 @@ void decompressionOutputCallback(void *decoderRef,
                                    rotation:RTCVideoRotation_0
                                 timeStampNs:CMTimeGetSeconds(timestamp) * rtc::kNumNanosecsPerSec];
   decodedFrame.timeStamp = decodeParams->timestamp;
-  decodeParams->callback(decodedFrame);
+  [decoder processFrame:decodedFrame reorderSize:decodeParams->reorderSize];
```

Comment:
if I understand `decompressionOutputCallback` is called on the decoder thread, but flush would be on a caller thread. It could mean the queue is accessed from 2 threads at once

Reply:
flush is calling VTDecompressionSessionWaitForAsynchronousFrames(), so it will block until all frames are given to `decompressionOutputCallback`. Once VTDecompressionSessionWaitForAsynchronousFrames() is completed, `decompressionOutputCallback` will not be called until we call again decode. So we can safely call _callback in flush().